services/ux-backend/handler/onboarding.go

```diff
@@ -0,0 +1,103 @@
+package handler
+
+import (
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/red-hat-storage/ocs-operator/v4/services/types"
+)
+
+func OnboardingTokenHandler(w http.ResponseWriter, r *http.Request) {
+	switch r.Method {
+	case "POST":
+
+		response, err := getOnboardingToken()
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+		//nolint
+		w.Write(response)
+
+	default:
+		fmt.Fprintf(w, "Sorry, only POST methods are supported.")
+	}
+}
+
+func getOnboardingToken() ([]byte, error) {
+
+	expirationHour := os.Getenv("ExpirationHour")
+	if expirationHour == "" {
+		expirationHour = "48"
+	}
+
+	expireHour, err := strconv.Atoi(expirationHour)
+	if err != nil {
+		return nil, err
+	}
+
+	data := types.OnboardingTicket{
+		ID:             uuid.New().String(),
+		ExpirationDate: time.Now().Add(time.Duration(expireHour) * time.Hour).Unix(),
+	}
+
+	payload, err := json.Marshal(data)
+	if err != nil {
+		return nil, err
+	}
+
+	encodedPayload := base64.StdEncoding.EncodeToString(payload)
+	// Before signing, we need to hash our message
+	// The hash is what we actually sign
+	msgHash := sha256.New()
+	_, err = msgHash.Write(payload)
+	if err != nil {
+		return nil, err
+	}
+
+	msgHashSum := msgHash.Sum(nil)
+	pemString, err := os.ReadFile("/etc/onboarding-private-key/key")
+	if err != nil {
+		return nil, err
+	}
+
+	// In order to generate the signature, we provide a random number generator,
+	// our private key, the hashing algorithm that we used, and the hash sum
+	// of our message
+	Block, _ := pem.Decode(pemString)
+	privateKey, err := x509.ParsePKCS1PrivateKey(Block.Bytes)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Why are we reading the private key file and decoding the private key every time we call generated token? 

Preferably load and decode it once into an in-memory scope and reuse it.

Reply:
Agreed (for now)